shuriken/include/shuriken/exceptions/invalidinstruction_exception.h

```diff
@@ -39,6 +40,18 @@ namespace exceptions {
             return _inst_size;
         }
     };
+    class LifterException : public std::exception {
```

Comment:
Is `LifterException` used in the same place as `InvalidInstructionException`? If not, move `LifterException` to another file, because remember every header file will be expanded wherever you use it, so we would be adding two exceptions when the code is expanded.

Reply:
i'll move it to lifter.h

Reply:
or you can move it to "lifter_exception.h" it's fine that

Reply:
sounds good to me

Reply:
it seems moving to lifter.h is better, everything that wants to use lifter will probably needs access to its exception definitions too. I moved them to lifter.h instead.